### PR TITLE
improve(ecs): add resource image

### PR DESCRIPTION
**Pull request by 956081712**

### CODE REVIEW

File: alicloud/resource_alicloud_image.go [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+	return &schema.Resource{
+		Create: resourceAliCloudImageCreate,
+		Read:   resourceAliCloudImageRead,
+		Update: resourceAliCloudImageUpdate,
+		Delete: resourceAliCloudImageDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
```
Comment (ddaddaddda) [inline alicloud/resource_alicloud_image.go]: remove this

File: alicloud/resource_alicloud_image.go [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+		}
+	}
+
+	// The snapshot cannot be a snapshot created before July 15, 2013 (inclusive)
+	if snapshotId, ok := d.GetOk("snapshot_id"); ok {
+		snapshot, err := ecsService.DescribeSnapshot(snapshotId.(string))
+		if err != nil {
+			return WrapError(err)
+		}
+		snapshotCreationTime, err := time.Parse("2006-01-02T15:04:05Z", snapshot.CreationTime)
```
Comment (ddaddaddda) [inline alicloud/resource_alicloud_image.go]: err

File: alicloud/resource_alicloud_image.go [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+		request.Tag = &imageTags
+	}
+	request.SnapshotId = d.Get("snapshot_id").(string)
+	request.ResourceGroupId = d.Get("resource_group_id").(string)
+	request.Platform = d.Get("platform").(string)
+	request.ImageName = d.Get("name").(string)
+	request.Description = d.Get("description").(string)
+	request.Architecture = d.Get("architecture").(string)
+
+	err := resource.Retry(DefaultTimeoutMedium*time.Second, func() *resource.RetryError {
```
Comment (ddaddaddda) [inline alicloud/resource_alicloud_image.go]: remove retry

File: alicloud/resource_alicloud_image.go [hunk 4]
```diff
@@ -0,0 +1,459 @@
...
+	}
+	if d.HasChange("name") {
+		d.SetPartial("name")
+		if imageName, ok := d.GetOk("name"); ok {
+			request.ImageName = imageName.(string)
+		}
+		update = true
+	}
+	if update {
+		err = resource.Retry(1*time.Minute, func() *resource.RetryError {
```
Comment (ddaddaddda) [inline alicloud/resource_alicloud_image.go]: remove retry

File: alicloud/resource_alicloud_image.go [hunk 5]
```diff
@@ -0,0 +1,459 @@
...
+		if object.Usage == "instance" && force.(bool) {
+			return WrapError(Error("The current image is used by other instances"))
+		}
+	}
+	request := ecs.CreateDeleteImageRequest()
+	request.RegionId = client.RegionId
+	request.ImageId = object.ImageId
+	request.Force = requests.NewBoolean(d.Get("force").(bool))
+
+	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
```
Comment (ddaddaddda) [inline alicloud/resource_alicloud_image.go]: remove retry

File: alicloud/resource_alicloud_image.go [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+	ecsService := EcsService{client}
+	object, err := ecsService.DescribeImageById(d.Id())
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return WrapError(err)
+	}
+	d.SetId(object.ImageId)
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_image.go]: There is no need to set id again and please remove the lien 263

File: alicloud/resource_alicloud_image.go [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+			return nil
+		}
+		return WrapError(err)
+	}
+	request := ecs.CreateDeleteImageRequest()
+	// Whether to perform forced deletion
+	if force, ok := d.GetOk("force"); ok {
+		if object.Usage == "instance" && !force.(bool) {
+			return WrapError(Error("The current image is used by other instances"))
+		}
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_image.go]: please remove the line 291 and just retain line 295

File: website/docs/r/image.html.markdown [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+---
+
+# alicloud\_image
+
+Creates a custom image. You can then use a custom image to create ECS instances (RunInstances) or change the system disk for an existing instance (ReplaceSystemDisk).
+
+-> **NOTE:**  If you want to create a template from an ECS instance, you can specify the instance ID (InstanceId) to create a custom image. You must make sure that the status of the specified instance is Running or Stopped. After a successful invocation, each disk of the specified instance has a new snapshot created.
+-> **NOTE:**  If you want to create a custom image based on the system disk of your ECS instance, you can specify one of the system disk snapshots (SnapshotId) to create a custom image. However, the specified snapshot cannot be created on or before July 15, 2013.
+-> **NOTE:**  If you want to combine snapshots of multiple disks into an image template, you can specify DiskDeviceMapping to create a custom image.
+-> **NOTE:**  Available in 1.63.0+
```
Comment (xiaozhu36) [inline website/docs/r/image.html.markdown]: please set it to 1.64.0

**Pull request merged**